### PR TITLE
[E2E] Add search tests 4667

### PR DESCRIPTION
**Pull request by dimitriharding**
Relates to [#4667](https://github.com/nasa/openmct/issues/4667) 

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [ ] Command line build passes?
* [ ] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: e2e/tests/search.e2e.spec.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        const searchResults = await page.locator(searchResultSelector);
+
+        // Verify that the search result/s correctly match the search query
+        expect(await searchResults.count()).toBe(2);
+        await expect(await searchResults.first()).toHaveText(folderName);
+        await expect(await searchResults.last()).toHaveText(folderName2);
+    });
+});
+
+async function createFolderObject(page, folderName) {
```
Comment (unlikelyzero) [inline e2e/tests/search.e2e.spec.js:104]: Awesome 😎

File: e2e/tests/search.e2e.spec.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    await page.locator('text=Properties Title Notes >> input[type="text"]').click();
+
+    // Enter folder name
+    await page.locator('text=Properties Title Notes >> input[type="text"]').fill(folderName);
+
+    // Create folder object
+    await page.locator('text=OK').click();
+}
+
+async function waitForSearchCompletion(page) {
```
Comment (unlikelyzero) [inline e2e/tests/search.e2e.spec.js:121]: This is great. Another alternative is to .waitforselector

**Review by unlikelyzero — COMMENTED** [review]
Perfect! Just waiting on the CLA :)

File: e2e/tests/search.e2e.spec.js (hunk 2 above)
Comment (dimitriharding) [reply e2e/tests/search.e2e.spec.js:121]: Thanks for the feedback. Updated accordingly.

**Comment by unlikelyzero** [comment]
@dimitriharding this is great. If you want to begin on another test, please feel free!

I've updated the title of this PR to prevent someone from accidentally merging it before the CLA process has finished.

**Comment by unlikelyzero** [comment]
@dimitriharding you're now good to go! You'll need to rebase to pick up quite a few changes.

**Comment by dimitriharding** [comment]
> @dimitriharding you're now good to go! You'll need to rebase to pick up quite a few changes.

Should be updated now!

**Comment by unlikelyzero** [comment]
@dimitriharding would you mind running the following locally

```
npm run test:e2e:local -- search --repeat-each=5
```

**Review by unlikelyzero — APPROVED** [review]
🚀

**Comment by dimitriharding** [comment]
@unlikelyzero Just getting to fix the failing tests I saw when I ran the command, and looks like this was merged in earlier on. Here are the fixes: https://github.com/nasa/openmct/pull/5552

**Pull request merged**